ChattoAdditions/Tests/Chat Items/BaseMessage/CompoundMessagePresenterTests.swift

```diff
@@ -61,4 +61,78 @@ final class CompoundMessagePresenterTests: XCTestCase {
         XCTAssertEqual(0, presenter.invokedUpdateContentCount)
         XCTAssertEqual(1, presenter.invokedUpdateExistingContentPresentersCount)
     }
+
+    func test_GivenFailableContentPresenter_WhenItsContentIsFailedToLoad_ThenViewModelStatusUpdatedToFailed() throws {
+        let contentTransferStatus = Observable<TransferStatus>(.idle)
+        let (presenter, viewModel) = try self.makeRealPresenter(contentTransferStatus: contentTransferStatus)
+        contentTransferStatus.value = .failed
+        XCTAssertEqual(viewModel.messageContentTransferStatus, .failed)
+        // to get rid of the warning about not used variable
+        XCTAssert(presenter === presenter)
```

Comment:
I think we can do something like this to avoid extra assertion:
```swift
let (_, viewModel) = try self.makeRealPresenter(contentTransferStatus: contentTransferStatus)
```

Reply:
Unfortunately, due to the fact that compound presenter propagates the action from the content presenter to the View Model, it should be retained until the test case finishes. I tried to wrap the code in the case in autorelease pool but the unused presenter still was deallocating right after the initialisation.
Maybe you can suggest some other way how to keep in memory until the end of the test?

Reply:
May be this would work (at the end of the test):
```swift
_ = presenter
```